sirepo/template/webcon.py

```diff
@@ -356,6 +356,205 @@ def get_settings_report(run_dir, data):
+        ]
+        for cfg in y_info:
+            code.append(f"{cfg.y_var} = \
+            {data_var}[:, {cfg.y_index}][{x_range_inds_var}[0]:{x_range_inds_var}[1]]")
+        nb.add_code_cell(code)
+        if j == 0:
+            # only do this once
+            nb.add_report(PKDict(
+                x_var=x_var,
+                x_label=x_label,
+                y_info=y_info,
+                title='Analysis Plot'
+            ))
+
+        if 'action' in rpt:
+            # reset y_var to 1st data set
+            y_var = 'y1'
+            nb.add_markdown_cell([f"## {rpt.action} Plot"])
+            if rpt.action == 'fft':
+                # only 1 curve (for now?)
+                w_var = 'w'
+                y_norm_var = 'y_norm'
+                nb.add_code_cell([
+                    f"{w_var}, {y_norm_var} = get_fft({x_var}, {y_var})"
+                ])
+                y_info = [PKDict(
+                    y_var=y_norm_var,
+                    y_label='',
+                    style='line'
+                )]
+                nb.add_report(PKDict(
+                    x_var=w_var,
+                    x_label=x_label,
+                    y_info=y_info,
+                    title='FFT'
+                ))
+            if rpt.action == 'fit':
+                x_fit_var = 'x_fit'
+                y_fit_var = 'y_fit'
+                y_max_var = 'y_fit_max'
+                y_min_var = 'y_fit_min'
+                p_vals_var = 'p_vals'
+                sigma_var = 'sigma'
+                eqn = rpt.fitEquation
+                prm = rpt.fitParameters
+                var = rpt.fitVariable
+                nb.add_code_cell([
+                    f"{x_fit_var}, {y_fit_var}, {y_min_var}, {y_max_var}, {p_vals_var},\
+                    {sigma_var} = fit_to_equation(\
+                        {x_var}[{x_range_inds_var}[0]:{x_range_inds_var}[1]],\
+                        y1, '{eqn}', '{var}', '{prm}\
+                    ')"
+                ])
+                y_info = [
+                    PKDict(y_var=y_var, y_label='Data', style='scatter'),
+                    PKDict(
+                        y_var=y_fit_var, x_points=x_fit_var, y_label='Fit', style='line'
+                    ),
+                    PKDict(
+                        y_var=y_max_var, x_points=x_fit_var, y_label='Max', style='line'
+                    ),
+                    PKDict(
+                        y_var=y_min_var, x_points=x_fit_var, y_label='Min', style='line'
+                    ),
+                ]
+                nb.add_report(PKDict(
+                    x_var=x_var,
+                    x_label='',
+                    y_info=y_info,
+                    title='Fit'
+                ))
+            if rpt.action == 'cluster':
+                clusters_var = 'clusters'
+                cols = [idx for idx, f in enumerate(rpt.clusterFields) if f and \
+                        idx < len(col_info.names)]
+                cfg = PKDict(
```

Comment:
Why not leave these names the same? Then you wouldn't maybe need cfg?

Reply:
6af47c5

Redid all of this